frontend/catalyst/api_extensions/control_flow.py

```diff
@@ -549,7 +549,8 @@ def main():
     """
 
     def __init__(self, pred, true_fn):
-        self.preds = [pred]
+        # Make sure the predicate is a boolean
+        self.preds = [jnp.astype(pred, bool)]
```

Comment:
Can you perform some more stringent verification of the input, including shape?

Reply:
In progress

Reply:
Done!